### PR TITLE
EgressIP: Reload certificates for the grpc heatlhcheck server

### PR DESCRIPTION
**Pull request by kyrtapz**
Reload the certificates for the egressip grpc heatlhcheck server to handle rotation.
This is not something we can easily verify upstream, I verified it [downstream](https://github.com/openshift/ovn-kubernetes/pull/2146)(4.12) by decreasing the certificate lifetime to 10min in CNO and checking that the certificate was properly reloaded by ovnkube-node:
```
# Ran in ovnkube-master container

openssl s_client -connect 10.129.0.2:9107| openssl x509 -noout -dates
Can't use SSL_get_servername
depth=1 CN = openshift-ovn-kubernetes_ovn-ca@1714473143
verify error:num=19:self signed certificate in certificate chain
verify return:1
depth=1 CN = openshift-ovn-kubernetes_ovn-ca@1714473143
verify return:1
depth=0 CN = ovn
verify return:1
notBefore=Apr 30 10:53:42 2024 GMT
notAfter=Apr 30 11:03:43 2024 GMT

#Wait a couple of minutes...

openssl s_client -connect 10.129.0.2:9107| openssl x509 -noout -dates
Can't use SSL_get_servername
depth=1 CN = openshift-ovn-kubernetes_ovn-ca@1714473143
verify error:num=19:self signed certificate in certificate chain
verify return:1
depth=1 CN = openshift-ovn-kubernetes_ovn-ca@1714473143
verify return:1
depth=0 CN = ovn
verify return:1
notBefore=Apr 30 10:58:46 2024 GMT
notAfter=Apr 30 11:08:47 2024 GMT
```

### CODE REVIEW

**Comment by abhat** [comment]
@kyrtapz should we be adding a e2e test around cert rotation?

**Comment by kyrtapz** [comment]
> @kyrtapz should we be adding a e2e test around cert rotation?

We do not manage/setup certificates for ovn-kubernetes upstream.

File: go-controller/pkg/ovn/healthcheck/egressip_healthcheck.go [hunk 1]
```diff
@@ -70,13 +73,29 @@ func (ehs *egressIPHealthServer) Run(stopCh <-chan struct{}) {
 	if cfg.Cert == "" || cfg.PrivKey == "" {
 		klog.Warning("Health checking using insecure connection")
 	} else {
-		// Enable TLS for all incoming connections.
-		cert, err := tls.LoadX509KeyPair(cfg.Cert, cfg.PrivKey)
+		// certProvider is responsible for reloading the certificate if it rotates.
+		certProvider, err := pemfile.NewProvider(pemfile.Options{
+			CertFile:        cfg.Cert,
+			KeyFile:         cfg.PrivKey,
+			RefreshDuration: time.Minute,
```
Comment (jcaamano) [inline go-controller/pkg/ovn/healthcheck/egressip_healthcheck.go:80]: Nothing against 1 minute, but what's the reasoning for the 1 hour default not being appropriate?

Comment (kyrtapz) [reply go-controller/pkg/ovn/healthcheck/egressip_healthcheck.go:80]: It is a corner case scenario but this is going to help with suspended clusters...
See: https://github.com/ovn-org/ovn-kubernetes/pull/4118 for a similar issue.

Comment (jcaamano) [reply go-controller/pkg/ovn/healthcheck/egressip_healthcheck.go:80]: hmm but that was specific to cert manager and the mechanism it used to wait for rotation. How does that apply here?

**Review by jcaamano — APPROVED** [review]

**Pull request merged**